File: java/oop/CustomerApp/src/com/techlabs/customer/Customer.java
package com.techlabs.customer;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Customer {
	
	public static int tempId;
	private int id;
	private String firstname;
	private String lastname;
	private Date dateOfBirth;
	
	static{
		tempId=100;
	}
	
	public Customer(String firstname,String lastname,String dob)throws ParseException{
		tempId++;
		this.id=tempId;
		this.firstname=firstname;
		this.lastname=lastname;
		this.dateOfBirth=convertToDate(dob);
	}
	
	public int getId(){
		return this.id;
	}
	
	public String getFirstname(){
		return this.firstname;
	}
	
	public String getLastname(){
		return this.lastname;
	}
	
	public String getDateOfBirth(){
		SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
		return formatter.format(dateOfBirth);
	}
	
	private Date convertToDate(String dob)throws ParseException{
		SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
		Date date=null;
		date = formatter.parse(dob);
		return date;
	}
	
	public int calculateAge(){
		Date date=new Date();
		SimpleDateFormat formatter=new SimpleDateFormat("yyyy");
		int curr_year = Integer.parseInt(formatter.format(date));
		int dob_year=Integer.parseInt(formatter.format(this.dateOfBirth));
		
		return curr_year-dob_year;
	}
}

File: dotNet/CSharp/CustomerApp/CustomerApp/Customer.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Text;

namespace CustomerApp
{
    public class Customer
    {
        private static int _tempId;
        private int _id;
        private string _firstname;
        private string _lastname;
        private DateTime _dateOfBirth;

        static Customer()
        {
            _tempId = 100;
        }

        public Customer(string firstName,string lastName,string dateOfBirth)
        {
            _tempId++;
            _id = _tempId;
            _firstname = firstName;
            _lastname = lastName;
            _dateOfBirth = ConvertToDate(dateOfBirth);
            CalculateAge();
        }

        public Int32 Id
        {
            get
            {
                return _id;
            }
        }

        public String FirstName
        {
            get
            {
                return _firstname;
            }
        }

        public String LastName
        {
            get
            {
                return _lastname;
            }
        }

        public String DateOfBirth
        {
            get
            {
                return _dateOfBirth.ToLongDateString();
            }
        }

        private DateTime ConvertToDate(string date)
        {
            return DateTime.Parse(date);
        }

        public Int32 CalculateAge()
        {
            DateTime date = DateTime.Now;
            Int32 curr_year = date.Year;
            Int32 dob_year = _dateOfBirth.Year;

            return curr_year-dob_year;
        }

        public override string ToString()
        {
            return "Customer ID: "+_id
                +"\n First Name: "+_firstname
                +"\n Last Name: "+_lastname
                +"\n Date Of Birth: "+DateOfBirth
                +"\n Age: "+CalculateAge();
        }
    }
}
